dist/secure-handlebars.js

```diff
@@ -10141,5 +10055,699 @@ module.exports.create = overrideHbsCreate;
+*
+* @description
+* create a new parser with all internal states inherited
+*/
+StrongContextParser.prototype.fork = function() {
+    var parser = new this.constructor(this.config, this.listeners);
+
+    parser.state = this.state;
+    parser.tagNames = this.tagNames.slice();
+    parser.tagNameIdx = this.tagNameIdx;
+    parser.attributeName = this.attributeName;
+    parser.attributeValue = this.attributeValue;
+
+    if (this.config.enableStateTracking) {
+        parser.buffer = this.buffer.slice();
+        parser.states = this.states.slice();
+    }
+
+    return parser;
+};
+
+StrongContextParser.prototype.on = function(eventType, listener) {
+    var self = this, listeners = self.listeners[eventType];
+    if (listener) {
+        if (listeners) {
+            listeners.push(listener);
+        } else {
+            self.listeners[eventType] = [listener];
+        }
+    }
+    return self;
+};
+
+StrongContextParser.prototype.once = function(eventType, listener) {
+    var self = this, onceListener;
+    if (listener) {
+        onceListener = function () {
+            self.off(eventType, onceListener);
+            listener.apply(self, arguments);
+        };
+        return this.on(eventType, onceListener);
+    }
+    return self;
+};
+
+StrongContextParser.prototype.off = function (eventType, listener) {
+    if (listener) {
+        var i, len, listeners = this.listeners[eventType];
+        if (listeners) {
+            for (i = 0; listeners[i]; i++) {
+                if (listeners[i] === listener) {
+                    listeners.splice(i, 1);
+                    break;
+                }
+            }
+        }
+    }
+    return this;
+};
+
+StrongContextParser.prototype.emit = function (eventType) {
+    var self = this,
+        listeners = self.listeners[eventType],
+        i, args, listener;
+
+    if (listeners) {
+        args = [].slice.call(arguments, 1);
+        for (i = 0; (listener = listeners[i]); i++) {
+            listener.apply(self, args);
+        }
+    }
+    return self;
+};
+
+/**
+ * @function StrongContextStrongContextParser#parsePartial
+ *
+ * @param {string} input - The HTML fragment
+ * @returns {string} the inputs with parse errors and browser-inconsistent characters automatically corrected
+ *
+ * @description
+ * <p>Perform HTML fixer before the contextual analysis</p>
+ *
+ */
+StrongContextParser.prototype.parsePartial = function(input, endsWithEOF) {
+    var self = this;
+    self.input = input.split('');
+    self.inputLen = self.input.length;
+
+    for (var i = 0, lastState; i < self.inputLen; i++) {
+        lastState = self.state;
+
+        self.emit('preWalk', lastState, i, endsWithEOF);
+        self.walk(i, self.input, endsWithEOF);
+        self.emit('postWalk', lastState, self.state, i, endsWithEOF);
+    }
+
+    return (self.output = self.input.join(''));
+};
+
+
+StrongContextParser.prototype.walk = function(i, input, endsWithEOF) {
```

Comment:
i have some concern about the maintenance of this function, i understand that u r having the similar logic with the original walk function with different function signature. we should have next discussion to absorb it back to original CP.